src/components/Footer.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { Dispatch } from 'react';
+import { Todo } from '../types/Todo';
+import { FilterTypes } from '../App';
+import cn from 'classnames';
+
+interface Props {
+  todos: Todo[];
+  onSettingFilter: Dispatch<React.SetStateAction<FilterTypes>>;
+  filterType: FilterTypes;
+}
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  onSettingFilter,
+  filterType,
+}) => {
+  const amountOfActiveTodos = todos.filter(todo => !todo.completed).length;
+  const isSomeTodoIsCompleted = todos.some(todo => todo.completed === true);
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${amountOfActiveTodos} items left`}
+      </span>
+
+      {/* Active link should have the 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        <a
+          onClick={() => onSettingFilter(FilterTypes.All)}
+          href="#/"
+          className={cn('filter__link', {
+            selected: filterType === FilterTypes.All,
+          })}
+          data-cy="FilterLinkAll"
+        >
+          All
+        </a>
```

Comment:
You can generate an array from enum with `Object.keys/values/entries` and map thought it to generate links